### PR TITLE
Hec 1141 mongo repo change

### PR DESCRIPTION
**Pull request by priyankadhiman3006**
mongo repo changes

### CODE REVIEW

File: app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala [hunk 1]
```diff
@@ -16,125 +16,79 @@
...
-import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.{ExecutionContext, Future}
 
 @ImplementedBy(classOf[SessionStoreImpl])
 trait SessionStore {
 
   def get()(implicit
-    hc: HeaderCarrier
+    hc: HeaderCarrier,
```
[the remark this answers is not in the document]
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala:37]: yup, forgot to removed it.

[the remark this answers is not in the document]
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala]: updated

File: app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala [hunk 2]
```diff
@@ -16,125 +16,79 @@
...
-                         )
-                     )
-          } yield result
-
-          response.value
-        }
-        .recover { case e ⇒ Left(Error(e)) }
-    }
+  )(implicit hc: HeaderCarrier, request: Request[_]): EitherT[Future, Error, Unit] =
+    EitherT(doStore(sessionData))
```
Comment (meghana-viswanath) [inline app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala:65]: that's a large amount of simplification! given all we are doing in the `store` and `get` methods is to call out the the `doStore/Get` methods, can we just move the content of those methods directly into `store` and `get`?

File: app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala [hunk 3]
```diff
@@ -16,125 +16,70 @@
...
-      .get[FiniteDuration]("session-store.expiry-time")
-      .value
-
-    new CacheMongoRepository("sessions", expireAfter.toSeconds)(
-      mongo.mongoConnector.db,
-      ec
+) extends SessionCacheRepository(
+      mongoComponent = mongo,
+      collectionName = "sessions",
+      ttl = configuration.get[FiniteDuration]("session-store.ttl"),
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala:51]: this should point to the existing `session-store.expiry-time` value - 150 days is for the BE mongo, this is just for session data in the FE

Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala]: updated

File: app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala (hunk 2 above)
Comment (priyankadhiman3006) [reply app/uk/gov/hmrc/heclicensingbodyfrontend/repos/SessionStore.scala]: done

**Review by meghana-viswanath — APPROVED** [review]

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**